diff --git a/node_modules/expo-av/android/build.gradle b/node_modules/expo-av/android/build.gradle
index 749f6bdfd9fb06b33047a56859b57fe18699c391..f810e10b1e7c017e877d37b7e5b7dd5bd141b67e 100644
--- a/node_modules/expo-av/android/build.gradle
+++ b/node_modules/expo-av/android/build.gradle
@@ -82,7 +82,7 @@
 dependencies {
   implementation 'com.facebook.react:react-android'
 
-  compileOnly 'com.facebook.fbjni:fbjni:0.3.0'
+  implementation 'com.facebook.fbjni:fbjni:0.6.0'

   api 'com.google.android.exoplayer:exoplayer:2.18.1'
   api 'com.google.android.exoplayer:extension-okhttp:2.18.1'
